File: 0748-largest-number-at-least-twice-of-others/0748-largest-number-at-least-twice-of-others.py
class Solution:
    def dominantIndex(self, nums: List[int]) -> int:
        max_num = max(nums)
        max_index = nums.index(max_num)
        for i in nums:
            if i != max_num and i*2 > max_num:
                return -1 
        return max_index
        

File: 0748-largest-number-at-least-twice-of-others/0748-largest-number-at-least-twice-of-others.java
class Solution {
    public int dominantIndex(int[] nums) {
        int max = Integer.MIN_VALUE;
        int max_index = Integer.MIN_VALUE;
        for(int i = 0;i<nums.length;i++){
            if (nums[i] > max) {
                max = nums[i];
                max_index = i;
            }
        }
        for(int i = 0;i<nums.length;i++){
            if((nums[i] != max) && (nums[i]*2 > max)){
                return -1;
            }
        }
        return max_index;
    }
}